miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt

```diff
@@ -3,18 +3,31 @@ package com.rakuten.tech.mobile.miniapp
 import androidx.annotation.VisibleForTesting
 import com.rakuten.tech.mobile.miniapp.api.ApiClient
 import com.rakuten.tech.mobile.miniapp.api.ManifestEntity
+import com.rakuten.tech.mobile.miniapp.storage.MiniAppStatus
 import com.rakuten.tech.mobile.miniapp.storage.MiniAppStorage
 
 internal class MiniAppDownloader(
     val storage: MiniAppStorage,
-    val apiClient: ApiClient
+    val apiClient: ApiClient,
+    val miniAppStatus: MiniAppStatus
 ) {
 
+    suspend fun getMiniApp(appId: String, versionId: String): String {
+        if (miniAppStatus.isVersionDownloaded(appId, versionId)) {
+            val baseSavePath = storage.getSavePathForApp(appId, versionId)
+            if (storage.filePathExists(baseSavePath))
```

Comment:
What is the scenario when version is downloaded but the path doesn't exist?

Reply:
When someone deletes data folder in local storage.

Reply:
I think shared preferences are stored in the same place. Isn't it?

Reply:
The SharedPreference just saves the state of the miniapp so need to check if the data source code of mini app whether it's still exsited.